lib/Backend/IRBuilder.cpp

```diff
@@ -1824,6 +1825,13 @@ IRBuilder::BuildReg2(Js::OpCode newOpcode, uint32 offset, Js::RegSlot R0, Js::Re
         this->AddInstr(instr, offset);
         return;
     }
+
+    // In order to obtain this instr from NewScObj, we must assure that, while walking
+    // up the ArgOut chain, this instr is accessible via instrDef and thus this instr's
+    // dst cannot have a symbol that has already been defined.
+    case Js::OpCode::GenCtorObj:
+        SetTempUsed(R0, TRUE);
```

Comment:
any reason why this isn't the same as case for `Js::OpCode::UpNewScObjCache`?

Reply:
What is SetTempUsed accomplishing here? R0 is not a temp and shouldn't be renumbered regardless.

Reply:
It's odd that we should have to do this. What does the byte code sequence look like? Where is the reg defined?

Reply:
Do you have any input on this question? Thanks.

Reply:
This hack is necessary for the following scenario:
- A dest symbol that has a def but no use can be emitted (ex: "a = b + c" where "a" is never used)
- "a" will be given a symbol (ex: "s5") but the next time a dest symbol is required (ex: a GenCtorObj instr) "s5" will be reused since "a" is never used.
- now "s5" is the dest of two instrs.
- "s5" loses its instrDef value because it has more than one instrDef

We need instrDef in order to find the GenCtorObj instr during the ArgOut chain traversal. so to solve this issue we make sure that the symbol of a GenCtorObj instr is unique.

This is unnecessary for UpNewScObjCache because UpNewScObjCache is not involved in any ArgOut chain traversal.

Reply:
Can you give me a script that exhibits the behavior you're talking about? If an unused dst is not consumed at the def point, we'll have more problems than just the one you're describing.

Reply:
@pleath Looks like we ran into this issue for `OpCode::StArrInlineItem_CI4` in the past (but handled it differently). Based on comment for that opcode in IRBuilder, I was able to find simple case that doesn't consume the dst: `for (let i = 0; i < 0; i + 1) { }`

Reply:
Arf. Do you have a little script that behaves this way? We need an automated way of detecting these. Maybe a debug flag that indicates a parse node is participating in a question expression.

Reply:
Now that I think of it, a hint from the parser to identify intermediate results of ?: ops might allow us to simplify all these things and eliminate Unused altogether. That's probably the way to go, not least because there's nothing that requires Unused for correctness, so we'll keep getting these unrenamed lifetimes. I'll look into that.